File: implementation/scipamato/public/web/src/test/java/ch/difty/scipamato/publ/web/common/BasePanelTest.java
package ch.difty.scipamato.publ.web.common;

import static org.assertj.core.api.Assertions.assertThat;

import org.apache.wicket.model.Model;
import org.junit.jupiter.api.Test;

import ch.difty.scipamato.common.web.Mode;
import ch.difty.scipamato.publ.web.WicketTest;

class BasePanelTest extends WicketTest {

    private BasePanel<String> panel;

    @Test
    void instantiatingWithIdOnly() {
        panel = new BasePanel<>("panel") {
            private static final long serialVersionUID = 1L;
        };
        assertThat(panel.getLocalization()).isEqualTo("en_us");
    }

    @Test
    void instantiatingWithIdAndModel() {
        panel = new BasePanel<>("panel", Model.of("foo")) {
            private static final long serialVersionUID = 1L;
        };
        assertThat(panel.getLocalization()).isEqualTo("en_us");
    }

    @Test
    void instantiatingWithIdAndModelAndMode() {
        panel = new BasePanel<>("panel", Model.of("foo"), Mode.EDIT) {
            private static final long serialVersionUID = 1L;
        };
        assertThat(panel.getLocalization()).isEqualTo("en_us");
    }

}

File: public/public-web/src/test/kotlin/ch/difty/scipamato/publ/web/common/BasePanelTest.kt
package ch.difty.scipamato.publ.web.common

import ch.difty.scipamato.common.web.Mode
import ch.difty.scipamato.publ.web.WicketTest
import org.amshove.kluent.shouldBeEqualTo
import org.apache.wicket.model.Model
import org.junit.jupiter.api.Test

class BasePanelTest : WicketTest() {

    private lateinit var panel: BasePanel<String>

    @Test
    fun instantiatingWithIdOnly() {
        panel = object : BasePanel<String>("panel") {
            private val serialVersionUID: Long = 1L
        }
        panel.localization shouldBeEqualTo "en_us"
    }

    @Test
    fun instantiatingWithIdAndModel() {
        panel = object : BasePanel<String>("panel", Model.of("foo")) {
            private val serialVersionUID: Long = 1L
        }
        panel.localization shouldBeEqualTo "en_us"
    }

    @Test
    fun instantiatingWithIdAndModelAndMode() {
        panel = object : BasePanel<String>("panel", Model.of("foo"), Mode.EDIT) {
            private val serialVersionUID: Long = 1L
        }
        panel.localization shouldBeEqualTo "en_us"
    }
}
